### PR TITLE
fix: remove ipn via email #1241

### PR DESCRIPTION
**Pull request by escapedcat**
@Kukks maybe it's easier to just have a look at the PR which is removing all IPN-sending via Email. I can change it according to what you think. Thanks.

I would resolve any conflicts after your feedback.

### CODE REVIEW

File: BTCPayServer/Controllers/InvoiceController.cs [hunk 1]
```diff
@@ -81,7 +81,6 @@ internal async Task<DataWrapper<InvoiceResponse>> CreateInvoiceCore(CreateInvoic
             entity.FullNotifications = invoice.FullNotifications || invoice.ExtendedNotifications;
             entity.ExtendedNotifications = invoice.ExtendedNotifications;
             entity.NotificationURLTemplate = invoice.NotificationURL;
-            entity.NotificationEmail = invoice.NotificationEmail;
```
Comment (Kukks) [inline BTCPayServer/Controllers/InvoiceController.cs]: No, this needs to stay for bitpay compatibility

File: BTCPayServer/HostedServices/InvoiceNotificationManager.cs [hunk 2]
```diff
@@ -118,16 +118,6 @@ void Notify(InvoiceEntity invoice, InvoiceEvent invoiceEvent, bool extendedNotif
 #pragma warning restore CS0618
             }
 
-            if (!String.IsNullOrEmpty(invoice.NotificationEmail))
```
Comment (Kukks) [inline BTCPayServer/HostedServices/InvoiceNotificationManager.cs]: No, this needs to stay for bitpay compatibility

File: BTCPayServer/Models/CreateInvoiceRequest.cs [hunk 3]
```diff
@@ -56,8 +56,6 @@ public class CreateInvoiceRequest
         public string Currency { get; set; }
         [JsonProperty(PropertyName = "price", DefaultValueHandling = DefaultValueHandling.Ignore)]
         public decimal Price { get; set; }
-        [JsonProperty(PropertyName = "notificationEmail", DefaultValueHandling = DefaultValueHandling.Ignore)]
```
Comment (Kukks) [inline BTCPayServer/Models/CreateInvoiceRequest.cs]: No, this needs to stay for bitpay compatibility

File: BTCPayServer/Controllers/InvoiceController.cs (hunk 1 above)
Comment (escapedcat) [reply BTCPayServer/Controllers/InvoiceController.cs]: Alright, thanks for your feedback! I'll update this.

**Comment by escapedcat** [comment]
I guess there's more to fix to make the tests work.

File: BTCPayServer/Services/Invoices/InvoiceEntity.cs [hunk 4]
```diff
@@ -325,11 +325,6 @@ public bool FullNotifications
             get;
             set;
         }
-        public string NotificationEmail
```
Comment (Kukks) [inline BTCPayServer/Services/Invoices/InvoiceEntity.cs]: Can't remove this one.

File: BTCPayServer/Models/InvoicingModels/InvoiceDetailsModel.cs [hunk 5]
```diff
@@ -120,7 +120,6 @@ public ProductInformation ProductInformation
         public AddressModel[] Addresses { get; set; }
         public DateTimeOffset MonitoringDate { get; internal set; }
         public List<Data.InvoiceEventData> Events { get; internal set; }
-        public string NotificationEmail { get; internal set; }
```
Comment (Kukks) [inline BTCPayServer/Models/InvoicingModels/InvoiceDetailsModel.cs]: Can't remove this one

**Comment by escapedcat** [comment]
@Kukks looks ok now, please check again, thanks.

**Pull request merged**